From 0760ef067487f76c859124d5b250dfea81aca9b0 Mon Sep 17 00:00:00 2001
From: Adam YH Lee <adam.yh.lee@gmail.com>
Date: Mon, 8 Dec 2014 13:27:21 -0800
Subject: [PATCH 1/3] Gumstix Overo Arbor43C Device Tree

Tested Features
- Multitouch
- Reset button
- Ethernet

Arbor43c has these 3 gpio connected LEDs:

- GPIO 186 (yellow)
- GPIO 163 (red)
- GPIO 150 (blue)

They have been tested with the maximum brightness of value 255
through the kernel GPIO controller interface (/sys/class/leds).

GPIO 186 and 150 work. GPIO 163 does not work.

Arbor43c has 2 push buttons:

- nRESET
- GPIO114

The nRESET signal does not need to be configured. It resets the
circuitary.

GPIO114 is exposed and is tested with evtest.
---
 arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi |  69 ++++++++++
 arch/arm/boot/dts/omap3-overo-arbor43c.dts         |  38 ++++++
 arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi   | 149 +++++++++++++++++++++
 arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts   |  39 ++++++
 4 files changed, 295 insertions(+)
 create mode 100644 arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi
 create mode 100644 arch/arm/boot/dts/omap3-overo-arbor43c.dts
 create mode 100644 arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi
 create mode 100644 arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts

diff --git a/arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi b/arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi
new file mode 100644
index 0000000..d869480
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-arbor43c-common.dtsi
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * Arbor43C expansion board is manufactured by Gumstix Inc.
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+#include "omap3-overo-common-peripherals.dtsi"
+#include "omap3-overo-common-lcd43c.dtsi"
+
+#include <dt-bindings/input/input.h>
+
+/ {
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&led_pins>;
+		heartbeat {
+			label = "overo:red:gpio163";
+			gpios = <&gpio6 3 GPIO_ACTIVE_HIGH>;		/* gpio_163 */
+			/* linux,default-trigger = "heartbeat"; */
+		};
+		gpio150 {
+			label = "overo:blue:gpio150";
+			gpios = <&gpio5 22 GPIO_ACTIVE_HIGH>;		/* gpio_150 */
+		};
+		gpio186 {
+			label = "overo:yellow:gpio186";
+			gpios = <&gpio6 26 GPIO_ACTIVE_HIGH>; 	/* gpio_186 */
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&button_pins>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		button1@114 {
+			label = "button1";
+			linux,code = <BTN_1>;
+			gpios = <&gpio4 18 GPIO_ACTIVE_LOW>;		/* gpio_114 */
+			gpio-key,wakeup;
+		};
+	};
+};
+
+#include "omap-gpmc-smsc9221.dtsi"
+
+&gpmc {
+	ranges = <5 0 0x2c000000 0x1000000>;	/* CS5 */
+
+	ethernet@gpmc {
+		reg = <5 0 0xff>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <16 IRQ_TYPE_LEVEL_LOW>;	/* GPIO 176 */
+	};
+};
+
+&lis33de {
+	status = "disabled";
+};
+
diff --git a/arch/arm/boot/dts/omap3-overo-arbor43c.dts b/arch/arm/boot/dts/omap3-overo-arbor43c.dts
new file mode 100644
index 0000000..203e24a
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-arbor43c.dts
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * Arbor43C expansion board is manufactured by Gumstix Inc.
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+/dts-v1/;
+
+#include "omap3-overo.dtsi"
+#include "omap3-overo-arbor43c-common.dtsi"
+
+/ {
+	model = "OMAP35xx Gumstix Overo on Arbor43C";
+	compatible = "gumstix,omap3-overo-arbor43c", "gumstix,omap3-overo", "ti,omap3430", "ti,omap3";
+};
+
+&omap3_pmx_core2 {
+	led_pins: pinmux_led_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x21E0, PIN_OUTPUT | MUX_MODE4)	/* sys_nirq.gpio_186 */
+			OMAP3630_CORE2_IOPAD(0x2198, PIN_OUTPUT | MUX_MODE4)	/* mcbsp1_clkx.gpio_168 */
+			OMAP3630_CORE2_IOPAD(0x2180, PIN_OUTPUT | MUX_MODE4)	/* uart1_cts.gpio_150 */
+		>;
+	};
+
+	button_pins: pinmux_button_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x25dc, PIN_INPUT | MUX_MODE4)		/* csi2_dx1.gpio114 */
+		>;
+	};
+};
diff --git a/arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi b/arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi
new file mode 100644
index 0000000..59bc6c9
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-common-lcd43c.dtsi
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * 4.3'' LCD Capacitive panel output for some Gumstix Overo boards (Arbor43C)
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+&omap3_pmx_core {
+	dss_dpi_pins: pinmux_dss_dpi_pins {
+		pinctrl-single,pins = <
+			OMAP3_CORE1_IOPAD(0x20d4, PIN_OUTPUT | MUX_MODE0)	/* dss_pclk.dss_pclk */
+			OMAP3_CORE1_IOPAD(0x20d6, PIN_OUTPUT | MUX_MODE0)	/* dss_hsync.dss_hsync */
+			OMAP3_CORE1_IOPAD(0x20d8, PIN_OUTPUT | MUX_MODE0)	/* dss_vsync.dss_vsync */
+			OMAP3_CORE1_IOPAD(0x20da, PIN_OUTPUT | MUX_MODE0)	/* dss_acbias.dss_acbias */
+			OMAP3_CORE1_IOPAD(0x20dc, PIN_OUTPUT | MUX_MODE0)	/* dss_data0.dss_data0 */
+			OMAP3_CORE1_IOPAD(0x20de, PIN_OUTPUT | MUX_MODE0)	/* dss_data1.dss_data1 */
+			OMAP3_CORE1_IOPAD(0x20e0, PIN_OUTPUT | MUX_MODE0)	/* dss_data2.dss_data2 */
+			OMAP3_CORE1_IOPAD(0x20e2, PIN_OUTPUT | MUX_MODE0)	/* dss_data3.dss_data3 */
+			OMAP3_CORE1_IOPAD(0x20e4, PIN_OUTPUT | MUX_MODE0)	/* dss_data4.dss_data4 */
+			OMAP3_CORE1_IOPAD(0x20e6, PIN_OUTPUT | MUX_MODE0)	/* dss_data5.dss_data5 */
+			OMAP3_CORE1_IOPAD(0x20e8, PIN_OUTPUT | MUX_MODE0)	/* dss_data6.dss_data6 */
+			OMAP3_CORE1_IOPAD(0x20ea, PIN_OUTPUT | MUX_MODE0)	/* dss_data7.dss_data7 */
+			OMAP3_CORE1_IOPAD(0x20ec, PIN_OUTPUT | MUX_MODE0)	/* dss_data8.dss_data8 */
+			OMAP3_CORE1_IOPAD(0x20ee, PIN_OUTPUT | MUX_MODE0)	/* dss_data9.dss_data9 */
+			OMAP3_CORE1_IOPAD(0x20f0, PIN_OUTPUT | MUX_MODE0)	/* dss_data10.dss_data10 */
+			OMAP3_CORE1_IOPAD(0x20f2, PIN_OUTPUT | MUX_MODE0)	/* dss_data11.dss_data11 */
+			OMAP3_CORE1_IOPAD(0x20f4, PIN_OUTPUT | MUX_MODE0)	/* dss_data12.dss_data12 */
+			OMAP3_CORE1_IOPAD(0x20f6, PIN_OUTPUT | MUX_MODE0)	/* dss_data13.dss_data13 */
+			OMAP3_CORE1_IOPAD(0x20f8, PIN_OUTPUT | MUX_MODE0)	/* dss_data14.dss_data14 */
+			OMAP3_CORE1_IOPAD(0x20fa, PIN_OUTPUT | MUX_MODE0)	/* dss_data15.dss_data15 */
+			OMAP3_CORE1_IOPAD(0x20fc, PIN_OUTPUT | MUX_MODE0)	/* dss_data16.dss_data16 */
+			OMAP3_CORE1_IOPAD(0x20fe, PIN_OUTPUT | MUX_MODE0)	/* dss_data17.dss_data17 */
+			OMAP3_CORE1_IOPAD(0x2100, PIN_OUTPUT | MUX_MODE0)	/* dss_data18.dss_data18 */
+			OMAP3_CORE1_IOPAD(0x2102, PIN_OUTPUT | MUX_MODE0)	/* dss_data19.dss_data19 */
+			OMAP3_CORE1_IOPAD(0x2104, PIN_OUTPUT | MUX_MODE0)	/* dss_data20.dss_data20 */
+			OMAP3_CORE1_IOPAD(0x2106, PIN_OUTPUT | MUX_MODE0)	/* dss_data21.dss_data21 */
+			OMAP3_CORE1_IOPAD(0x2108, PIN_OUTPUT | MUX_MODE0)	/* dss_data22.dss_data22 */
+			OMAP3_CORE1_IOPAD(0x210a, PIN_OUTPUT | MUX_MODE0)	/* dss_data23.dss_data23 */
+		>;
+	};
+
+	lte430_pins: pinmux_lte430_pins {
+		pinctrl-single,pins = <
+			OMAP3_CORE1_IOPAD(0x2174, PIN_OUTPUT | MUX_MODE4)	/* uart2_cts.gpio_144 */
+		>;
+	};
+
+	backlight_pins: pinmux_backlight_pins {
+		pinctrl-single,pins = <
+			OMAP3_CORE1_IOPAD(0x2176, PIN_OUTPUT | MUX_MODE4)	/* uart2_rts.gpio_145 */
+		>;
+	};
+
+	ft5x0_pins: pinmux_ft5x0_pins {
+		pinctrl-single,pins = <
+			OMAP3_CORE1_IOPAD(0x2138, PIN_INPUT_PULLDOWN | MUX_MODE4)	/* csi2_dx1.gpio_114 */
+		>;
+	};
+};
+
+/* Needed to power the DPI pins */
+&vpll2 {
+	regulator-always-on;
+};
+
+&dss {
+	status = "ok";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&dss_dpi_pins>;
+
+	port {
+		dpi_out: endpoint {
+			remote-endpoint = <&lcd_in>;
+			data-lines = <24>;
+		};
+	};
+};
+
+/ {
+	aliases {
+		display0 = &lcd0;
+	};
+
+	lcd0: display@0 {
+		compatible = "samsung,lte430wq-f0c", "panel-dpi";
+		label = "lcd";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&lte430_pins>;
+		enable-gpios = <&gpio5 16 GPIO_ACTIVE_HIGH>;		/* gpio_144 */
+
+		port {
+			lcd_in: endpoint {
+				remote-endpoint = <&dpi_out>;
+			};
+		};
+
+		panel-timing {
+			clock-frequency = <9200000>;
+			hactive = <480>;
+			vactive = <272>;
+			hfront-porch = <8>;
+			hback-porch = <4>;
+			hsync-len = <41>;
+			vback-porch = <2>;
+			vfront-porch = <4>;
+			vsync-len = <10>;
+
+			hsync-active = <0>;
+			vsync-active = <0>;
+			de-active = <1>;
+			pixelclk-active = <1>;
+		};
+	};
+
+	backlight {
+		compatible = "gpio-backlight";
+		
+		pinctrl-names = "default";
+		pinctrl-0 = <&backlight_pins>;
+		gpios = <&gpio5 17 GPIO_ACTIVE_HIGH>;		/* gpio_145 */
+
+		default-on;
+	};
+};
+
+&i2c3 {
+		pinctrl-names = "default";
+		pinctrl-0 = "&i2c3_pins";
+		clock-frequency = <100000>;
+		status = "okay";
+
+		ft5x0x: ft5x0@38 {
+			compatible = "focaltech,ft5x0x";
+			reg = <0x38>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&ft5x0_pins>;
+			interrupt-parent = <&gpio1>;
+			interrupts = <10 0>; /* gpio_10 */
+			wake-gpios = <&gpio1 31 0>;	/* gpio_31 */
+		};
+	};
diff --git a/arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts b/arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts
new file mode 100644
index 0000000..6c03491
--- /dev/null
+++ b/arch/arm/boot/dts/omap3-overo-storm-arbor43c.dts
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2014 Gumstix, Inc. - https://www.gumstix.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/*
+ * Arbor43C expansion board is manufactured by Gumstix Inc.
+ * https://store.gumstix.com/index.php/products/635/
+ */
+
+/dts-v1/;
+
+#include "omap3-overo-storm.dtsi"
+#include "omap3-overo-arbor43c-common.dtsi"
+
+/ {
+	model = "OMAP36xx/AM37xx/DM37xx Gumstix Overo on Arbor43C";
+	compatible = "gumstix,omap3-overo-arbor43c", "gumstix,omap3-overo", "ti,omap36xx", "ti,omap3";
+};
+
+&omap3_pmx_core2 {
+	led_pins: pinmux_led_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x21E0, PIN_OUTPUT | MUX_MODE4)	/* sys_nirq.gpio_186 */
+			OMAP3630_CORE2_IOPAD(0x2198, PIN_OUTPUT | MUX_MODE4)	/* mcbsp1_clkx.gpio_168 */
+			OMAP3630_CORE2_IOPAD(0x2180, PIN_OUTPUT | MUX_MODE4)	/* uart1_cts.gpio_150 */
+		>;
+	};
+
+	button_pins: pinmux_button_pins {
+		pinctrl-single,pins = <
+			OMAP3630_CORE2_IOPAD(0x25dc, PIN_INPUT | MUX_MODE4)		/* csi2_dx1.gpio114 */
+		>;
+	};
+};
+
-- 
1.9.1

